admin.py

```diff
@@ -10,40 +11,44 @@
 from lib import errors
 from lib.utils import clean_json, error, expect_json, success
 
-app = Flask(__name__)
-app.config["SECRET_KEY"] = "ttmadmin"
-CORS(app)
+admin = Flask(__name__)
+admin.config["SECRET_KEY"] = "ttmadmin"
+CORS(admin)
 mdb = MongoClient(db.MONGO_URL).db
 
-@app.route("/auth", methods=["POST"])
-@expect_json(password=str)
-def admin_auth(body):
-    if(body['password'] == "ttmadmin"):
-        return success("Authorized")
-    return error(403, "Incorrect password")
+def requires_auth(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        token = request.headers.get('authorization')
+        if(token != "ttmadmin"):
+            return error(403, "Unauthorized")
+        return f(*args, **kwargs)
+    return decorated_function
 
-@app.route("/reports", methods=["GET"])
+@admin.route("/reports", methods=["GET"])
+@requires_auth
 def get_reports():
     reports = mdb.reports.find()
     data=[]
     for x in reports:
         x['_id'] = str(x['_id'])
         data.append(x)
     jsonData=json.dumps(data)
-    return jsonData
+    return success(jsonData)
 
-@app.route("/reportmessages", methods=["POST"])
-@expect_json(room_id=str)
-def get_reported_messages(body):
-    conversation = mdb.reported_messages.find({'room_id' : body['room_id']})
+@admin.route("/reports/<room_id>/messages", methods=["GET"])
+@requires_auth
+def get_reported_messages(room_id):
+    conversation = mdb.reported_messages.find({'room_id' : room_id})
     data=[]
     for x in conversation:
         x['_id'] = str(x['_id'])
         data.append(x)
     jsonData=json.dumps(data)
-    return jsonData
+    return success(jsonData)

```

Comment:
Actually this should probably just be `success(data)` - `success()` automatically dumps the dict to JSON (and same below)